src/utils/isSubsetOf.ts

```diff
@@ -0,0 +1,39 @@
+/** Taken from TypeScript collection lib to perfectly match the .isSubsetOf signature */
+export interface ReadonlySetLike<T> {
```

Comment:
nit: But do we actually use it on setlikes? If it's always another set, I'd rather leave it at `other: Set<T>`.

Reply:
Regardless of what we use that on, the method signature in Node uses setlike, thus this function's signature replicating that method must also use setlikes. Or so I believe, at least